src/mlang/backend_ir/bir.ml

```diff
@@ -95,17 +95,26 @@ type variable_data = variable Mir.variable_data_
 
 type function_name = string
 
-type rule = { rule_id : rule_id; rule_name : string; rule_stmts : stmt list }
+type rule_or_verif_code = Rule of stmt list | Verif of stmt
+
+and rule_or_verif = {
+  rov_id : rov_id;
+  rov_name : string Pos.marked;
+  rov_code : rule_or_verif_code;
+}
 
 and stmt = stmt_kind Pos.marked
 
 and stmt_kind =
   | SAssign of variable * variable_data
   | SConditional of expression * stmt list * stmt list
   | SVerif of condition_data
-  | SRuleCall of rule_id
+  | SRuleCall of rov_id
```

Comment:
I'm a bit confused here: is `rov_id` always an identifier for a `Rule`, or can it be a `Verif` too?

Reply:
Ah, that's a miss. Will patch